lib/matplotlib/traitlets.py

```diff
@@ -0,0 +1,374 @@
+        if old != new:
+            self._memory[key] = old
+        super(exdict, self).__setitem__(key, new)
+
+    def update(self, *args, **kwargs):
+        if len(args) > 1:
+            raise TypeError("update expected at most 1 arguments, got %d" % len(args))
+        other = dict(*args, **kwargs)
+        for key in other:
+            self[key] = other[key]
+
+    def setdefault(self, key, value=None):
+        if key not in self:
+            self[key] = self._default_generator()
+        return self[key]
+
+    def ex(self, key):
+        try:
+            return self._memory[key]
+        except KeyError, e:
+            return self._default_generator(key)
+
+    def _default_generator(self, key): pass
+
+    def __getstate__(self):
+        d = self.__dict__.copy()
+        # remove unpickleable method
+        d['_default_generator'] = None
+        return d
+
+
+class PrivateMethodMixin(object):
+
+    def __new__(cls, *args, **kwargs):
+        inst = super(PrivateMethodMixin,cls).__new__(cls, *args, **kwargs)
+        inst._trait_values = exdict(inst._trait_values)
+        meth = lambda klass, key: getattr(klass, key).default_value
+        inst._trait_values._default_generator = MethodType(meth, cls)
+        return inst
+
+    def force_callback(self, name, cross_validate=True):
+        if name not in self.traits():
+            msg = "'%s' is not a trait of a %s class"
+            raise TraitError(msg % (name, self.__class__))
+
+        trait = self.traits()[name]
+
+        new = self._trait_values[name]
+        try:
+            old = self._trait_values.ex(name)
+        except KeyError:
+            trait = getattr(self.__class__, name)
+            old = trait.default_value
+
+        self._notify_trait(name, old, new)
+        if cross_validate:
+            trait = self._retrieve_trait(name)
+            # ignores value output
+            trait._cross_validate(self, new)
+
+    def private(self, name, value=Undefined):
+        trait = self._retrieve_trait(name)
+
+        if value is Undefined:
+            if hasattr(trait, '__base_get__'):
+                return trait.__base_get__(self)
+            return getattr(self, name)
+        else:
+            trait._cross_validation_lock = True
+            _notify_trait = self._notify_trait
+            self._notify_trait = lambda *args: None
+            setattr(self, name, value)
+            self._notify_trait = _notify_trait
+            trait._cross_validation_lock = False
+
+    def _retrieve_trait(self, name):
+        try:
+            trait = getattr(self.__class__, name)
+            if not isinstance(trait, BaseDescriptor):
+                msg = "'%s' is a standard attribute, not a trait, of a %s class"
+                raise TraitError(msg % (name, self.__class__))
+        except AttributeError:
+            msg = "'%s' is not a trait of a %s class"
+            raise TraitError(msg % (name, self.__class__))
+        return trait
+
```

Comment:
I'm having trouble with loading things after pickling. Right now, the `_default_generator` method for the "exdict" class I have for keeping track of old values is no longer there (as is expected since I removed it in `__getstate__`). I would expect the following bit of code in "PrivateMethodMixin" to fix this, however it instead raises an error claiming that `_rasterized` is no longer an attribute of a Rectangle instance. I suspect that this is because I'm overriding a `__setstate__` method somewhere else, but I'm not sure:

```python
    def __setstate__(self, state):
        meth = lambda klass, key: getattr(klass, key).default_value
        state['_trait_values']._default_generator = MethodType(meth, self.__class__)
        return state.update(self.__dict__)
```